config/example-menu.cfg

```diff
@@ -0,0 +1,926 @@
+#   command - same like 'item' but with gcode trigger
+#   input - same like 'command' but has value changing capabilities
+#   list - menu element container, with entry and exit gcode triggers
+#   vsdcard - same as 'list' but will append files from virtual sdcard
+#   deck - special container for custom screens (cards) has entry and exit gcode triggers.
+#   card - special content card for custom screens. Can only be used in 'deck'!
+
+#[menu item1]
+#type: item
+#   Type will determine menu item properties and behaviours:
+#name:
+#   This is mandatory attribute for every menu element.
+#   You can use Python output formatting for parameter and transform values.
+#   Quotes can be used in the beginning and end of name.
+#cursor:
+#   It allows to change cursor character for selected menu element.
+#   Default value is > [Optional]
+#width:
+#   This attribute accepts integer value. Element name is cut to this width. [Optional]
+#scroll:
+#   This attribute accepts static boolean value. You can use it together with 'width'.
+#   When this is enabled then names longer than width are scrolled back and forth.
+#   By default it's disabled. [Optional]
+#enable:
+#   This attribute accepts static boolean values and parameters (converted to boolean).
+#   It accepts multiple logical expressions. Values separated by comma will return True if all elements are true.
+#   Values on different lines will return True if any element is true.
+#   You can use logical negation by using character ! as parameter prefix.
+#parameter:
+#   This parameter accepts single float value or special variable.
+#   All available parameter variables can be listed by 'MENU DO=dump' gcode, menu itself must be running.
+#   This value is available for output formatting as {0}
+#transform:
+#   This attribute allows to transform parameter value to something else.
+#   More than one transformation can be added. Each transformation must be on separate line.
+#   These transformed values are available for output formatting as {1}..{x}
+#
+#   (fromLow,fromHigh,toLow,toHigh) - interpolate re-maps a parameter value from one range to another.
+#   Output value type is taken from toHigh. It can be int or float.
+#
+#   (e1,e2) - boolean chooser, will convert parameter value to boolean type (index 0 and 1),
+#   and will choose corresponding value from list.
+#
+#   [e1,e2,...] - int chooser, will convert parameter value to int type and will
+#   choose corresponding value by index from list.
+#
+#   int, float, bool, str, abs, bin, hex, oct, days, hours, minutes, seconds
+#   These will convert parameter value to special form.
+#   int,float,bool,str,abs,bin,hex and oct are python functions.
+#   days,hours,minutes,seconds will convert parameter value (it's taken as seconds) to time specific value
+#
+#   pure interger or float value is used as parameter value scaler/multiplier
+#
+#   {key:value,..} special dictionary chooser, parameter value cast type by first key type
```

Comment:
FYI, I struggled to understand the transform parameter.  The syntax does not give much hint to the invoked action.  I wonder if the transformations could be made more explicit - for example: convert interpolate to `iterpolate(fromLow,fromHigh,toLow,toHigh)`, boolean chooser to `chooser(e1, e2)`, multiply to `multiply(99.3)`, etc.

Alternatively, it might be possible to use the python ast module to do some very basic format string parsing - something like:
```
name: "temp is {0:.2f}".format(extruder0.target)
```
The config file must not contain a general purpose python parser, but I wonder if some basic format strings could be supported.  Just an idea.